graphics/managed_surface.cpp

```diff
@@ -486,15 +484,25 @@ void transBlitPixel(TSRC srcVal, TDEST &destVal, const Graphics::PixelFormat &sr
 		rDest = rSrc;
 		gDest = gSrc;
 		bDest = bSrc;
+		aDest = 0xff;
 	} else {
 		// Partially transparent, so calculate new pixel colors
+		destFormat.colorToARGB(destVal, aDest, rDest, gDest, bDest);
 		double alpha = (double)aSrc / 255.0;
 		rDest = static_cast<byte>((rSrc * alpha) + (rDest * (1.0 - alpha)));
 		gDest = static_cast<byte>((gSrc * alpha) + (gDest * (1.0 - alpha)));
 		bDest = static_cast<byte>((bSrc * alpha) + (bDest * (1.0 - alpha)));
+		
+		double sAlpha = (double)aSrc / 255.0;
+		double dAlpha = (double)aDest / 255.0;
+		double dAlphaRatio = dAlpha * (1.0 - sAlpha) / (sAlpha + dAlpha * (1.0 - sAlpha));
+		rDest = static_cast<uint8>(rSrc * sAlpha + rDest * dAlphaRatio);
+		gDest = static_cast<uint8>(gSrc * sAlpha + gDest * dAlphaRatio);
+		bDest = static_cast<uint8>(bSrc * sAlpha + bDest * dAlphaRatio);
```

Comment:
I think this is off due to parenthesis. This acts as:
bSrc * sAlpha + bDest * (dAlpha * (1.0 - sAlpha) / (sAlpha + dAlpha * (1.0 - sAlpha)))

instead of:
(bSrc * sAlpha + bDest * dAlpha * (1.0 - sAlpha)) / (sAlpha + dAlpha * (1.0 - sAlpha))

Reply:
Yes, you are right. I got my equations wrong here. I push a change, and hopefully they are correct now.